libraries/standard/mqtt/utest/mqtt_utest.c

```diff
@@ -1205,6 +1209,17 @@ void test_MQTT_Publish( void )
     status = MQTT_Publish( &mqttContext, &publishInfo, 0 );
     TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
 
+    /* Test that sending a publish without a payload succeeds. */
+    publishInfo.pPayload = NULL;
+    publishInfo.payloadLength = 0;
+    MQTT_SerializePublishHeader_ExpectAnyArgsAndReturn( MQTTSuccess );
+    MQTT_SerializePublishHeader_ReturnThruPtr_pHeaderSize( &headerSize );
+    status = MQTT_Publish( &mqttContext, &publishInfo, 0 );
+    TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
+    /* Restore the test payload and length. */
```

Comment:
This kind of resetting required and multiple lines of scrolling to find out the right set of values being tested indicates towards the unread-ability of tests. This is a much bigger issue and a separate discussion though - nothing is needed in this PR.

Reply:
Yes I agree. I agree that the test cases should be better separated.